### PR TITLE
Code review 2 patel3mo

### PR DESCRIPTION
**Pull request by MonilPatel03**
**Analysis of program**:The program does compiles and runs with no issues while cloning. Easy to understand, all the connectivity with different files and naming it was done nicely.

**Was the program available in UC Github on time?**: Yes

**Is the program documented/commented well enough for you to understand**? There were few comments, but due to the naming conventions of the classes and variables, it was easy to follow.

**Does the program compile**? Yes

**Rationale behind your changes** :  I changed few the cases of the variable names as they were mention min Kotlin Best practices.  

**Concepts that I learned**:
I learnt to put commas after the end of each item
I learnt what a trailing comma is
I learnt that property names start with uppercase
I learnt what Latienit var is
I learnt the correct use of uppercase/lowercase in right places
I learnt to write the argument with correct logics.
I referred to https://kotlinlang.org/docs/coding-conventions.html
and https://kotlinlang.org/docs/whatsnew12.html#multiplatform-projects-experimental

Commits on my project:
1. https://github.com/ajwhite00/Memrepo/commit/4f06d0c6f4436f6beea4fd390cb60df5c09d68bf
2. https://github.com/ajwhite00/Memrepo/commit/15c995728d24841275ae4e491839820b4f6c4542
3. https://github.com/ajwhite00/Memrepo/commit/c775c96cbe6c99ae8ad55288b5badb14af67ba1f

### CODE REVIEW

File: app/src/main/java/com/example/uptowncampus/MainActivity.kt [hunk 1]
```diff
@@ -49,15 +49,15 @@ class MainActivity : ComponentActivity() {
         super.onCreate(savedInstanceState)
         setContent {
             viewModel.fetchBuildings()
-            val buildings by viewModel.buildings.observeAsState(initial = emptyList())
-            val savedBuildings by viewModel.savedBuildings.observeAsState(initial = emptyList())
+            val Buildings by viewModel.buildings.observeAsState(initial = emptyList())
+            val SavedBuildings by viewModel.savedBuildings.observeAsState(initial = emptyList())
```
Comment (MonilPatel03) [inline app/src/main/java/com/example/uptowncampus/MainActivity.kt:53]: Property name should be in Upppercase

File: app/src/main/java/com/example/uptowncampus/MainActivity.kt [hunk 2]
```diff
@@ -49,15 +49,15 @@ class MainActivity : ComponentActivity() {
...
             UptownCampusTheme {
                 // A surface container using the 'background' color from the theme
                 Surface(
                     modifier = Modifier.fillMaxSize(),
                     color = MaterialTheme.colors.background
                 ) {
-                    BuildingName(buildings, savedBuildings, viewModel.selectedSavedBuilding)
+                    BuildingName(Buildings, SavedBuildings, viewModel.selectedSavedBuilding)
```
Comment (MonilPatel03) [inline app/src/main/java/com/example/uptowncampus/MainActivity.kt:60]: Property name should be in Uppercase

File: app/src/main/java/com/example/uptowncampus/MainActivity.kt [hunk 3]
```diff
@@ -301,15 +301,15 @@ class MainActivity : ComponentActivity() {
         }
     }
     private fun signIn() {
-        val providers = arrayListOf(
+        val Providers = arrayListOf(
```
Comment (MonilPatel03) [inline app/src/main/java/com/example/uptowncampus/MainActivity.kt:304]: property name should be in Uppercase

File: app/src/main/java/com/example/uptowncampus/MainActivity.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/com/example/uptowncampus/MainActivity.kt:53]: Oh - no.  lowerCamelCase is correct for variable names.  Leave this as it were.

File: app/src/main/java/com/example/uptowncampus/MainActivity.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/com/example/uptowncampus/MainActivity.kt:60]: See note above.  This is not correct.

File: app/src/main/java/com/example/uptowncampus/MainActivity.kt [hunk 4]
```diff
@@ -84,7 +84,7 @@ class MainActivity : ComponentActivity() {
                         building -> DropdownMenuItem(onClick = {
                             expanded = false
 
-                        if (building.buildingName == viewModel.NEW_BUILDING) {
+                        if (building.buildingName == viewModel.newbUILDING) {
```
Comment (discospiff) [inline app/src/main/java/com/example/uptowncampus/MainActivity.kt:87]: UPPERCASE_WITH_UNDERSCORES is correct for constants, which this appears to be.

File: app/src/main/java/com/example/uptowncampus/MainActivity.kt [hunk 5]
```diff
@@ -75,7 +75,7 @@ class MainActivity : ComponentActivity() {
                 }
                 .padding(8.dp),
                 horizontalArrangement = Arrangement.Center,
-                verticalAlignment = Alignment.CenterVertically
+                verticalAlignment = Alignment.CenterVertically,
```
Comment (discospiff) [inline app/src/main/java/com/example/uptowncampus/MainActivity.kt:78]: Either way is fine.  This does not impact technical debt.

File: app/src/main/java/com/example/uptowncampus/MainActivity.kt (hunk 3 above)
Comment (discospiff) [reply app/src/main/java/com/example/uptowncampus/MainActivity.kt:304]: Incorrect.  See above.

File: app/src/main/java/com/example/uptowncampus/MainViewModel.kt [hunk 6]
```diff
@@ -9,24 +9,23 @@ import androidx.lifecycle.ViewModel
...
 import kotlinx.coroutines.launch
-import kotlinx.coroutines.selects.select
 import org.json.JSONException
 
-class MainViewModel(var buildingService : IBuildingService = BuildingService()) : ViewModel() {
+@Suppress("UNNECESSARY_SAFE_CALL")
+class MainViewModel(private var BuildingService : IBuildingService = BuildingService()) : ViewModel() {
 
-    internal val NEW_BUILDING = "New Building"
+    internal val newbUILDING = "New Building"
```
Comment (discospiff) [inline app/src/main/java/com/example/uptowncampus/MainViewModel.kt:22]: Leave this as it were.  UPPER_CASE_WITH_UNDERSCORES is correct for constants.  This is similar to the NEW_SPECIMEN constant we made in our class demo.

File: app/src/main/java/com/example/uptowncampus/MainViewModel.kt [hunk 7]
```diff
@@ -9,24 +9,23 @@ import androidx.lifecycle.ViewModel
...
     var buildings: MutableLiveData<List<Building>> = MutableLiveData<List<Building>>()
 
     var savedBuildings: MutableLiveData<List<SavedBuildings>> = MutableLiveData<List<SavedBuildings>>()
     var selectedSavedBuilding by mutableStateOf(SavedBuildings())
 
-    private lateinit var firestore : FirebaseFirestore
+    private var firestore : FirebaseFirestore
```
Comment (discospiff) [inline app/src/main/java/com/example/uptowncampus/MainViewModel.kt:28]: Why remove lateinit?

**Comment by discospiff** [comment]
Don't merge this branch.

**Comment by iByrd** [comment]
Did not use this review

**Pull request closed without merging**